src/Analyzer.Core/Rules/BuiltInRules.json

```diff
@@ -18,6 +18,52 @@
       ]
     }
   },
+  {
+    "name": "AppService_FunctionApp_HTTPS",
+    "description": "Function App should only be accessible over HTTPS",
+    "recommendation": "Use HTTPS to ensure server/service authentication and protect data in transit from network layer eavesdropping attacks.",
+    "helpUri": "https://github.com/Azure/template-analyzer/docs/built-in-bpa-rules.md/#function-app-should-only-be-accessible-over-https",
+    "evaluation": {
+      "resourceType": "Microsoft.Web/sites",
+      "where": {
+        "path": "kind",
+        "regex": "^functionapp"
+      },
+      "allOf": [
+        {
+          "path": "properties.httpsOnly",
+          "equals": true
+        }
+      ]
+    }
+  },
+  {
+    "name": "AppServiceWebApp_HTTPS",
+    "description": "Web Application should only be accessible over HTTPS",
+    "recommendation": "Use HTTPS to ensure server/service authentication and protect data in transit from network layer eavesdropping attacks.",
+    "helpUri": "https://github.com/Azure/template-analyzer/docs/built-in-bpa-rules.md/#web-application-should-only-be-accessible-over-https",
+    "evaluation": {
+      "resourceType": "Microsoft.Web/sites",
+      "where": {
+        "anyOf": [
+          {
+            "path": "kind",
```

Comment:
Why is this checking if kind is specified?

Reply:
According to the schema, `kind` is not a required property, which means any Web App could bypass this check simply by not specifying the kind.  Web App seems to be the default value if this is not specified, so that case is captured in this rule.

Reply:
Another thing about default values, I know they can change between versions of the API... Likely not going to be the case for this one, but something we should keep in mind.